lib/srv/ingress/reporter.go

```diff
@@ -193,17 +216,17 @@ func (r *Reporter) ConnectionClosed(service string, conn net.Conn) {
 
 // ConnectionAuthenticated reports a new authenticated connection, AuthenticatedConnectionClosed must
 // be called when the connection is closed.
-func (r *Reporter) ConnectionAuthenticated(service string, conn net.Conn) {
+func (r *Reporter) ConnectionAuthenticated(service string, metadata string, conn net.Conn) {
```

Comment:
```suggestion
func (r *Reporter) ConnectionAuthenticated(service, metadata string, conn net.Conn) {
```
nit: Having two strings in a function signature makes it easy to swap them. I'd create a new type for at least one of them to get a compiler error when this happens.

Reply:
Fixed: `ServiceMetadata` is now a fixed three-element named tuple.